"Exit with code 1" (but the scripts exit with 0)

Running this script:

    "watch": "concurrently \"riot -w --ext html tags js/src/tags.js\" \"babel js/src --out-dir js/dist --watch\"",

If I break with `CTRL`+`C`: it returns an error

```
[1] babel js/src --out-dir js/dist --watch exited with code null
[0] riot -w --ext html tags js/src/tags.js exited with code null
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! xxx@1.0.0 watch: `concurrently "riot -w --ext html tags js/src/tags.js" "babel js/src --out-dir js/dist --watch"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the xxx@1.0.0 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

So this is a strange behavior : the subscripts exited with code null, but not concurrently (fresh install).

Is this the normal behavior?

Thank you for your work!

--- a/bin/concurrently.spec.js
+++ b/bin/concurrently.spec.js
@@ -1,10 +1,15 @@
 const readline = require('readline');
+const _ = require('lodash');
 const Rx = require('rxjs');
 const { buffer, map } = require('rxjs/operators');
 const spawn = require('spawn-command');
 
 const isWindows = process.platform === 'win32';
-const killExitCode = isWindows ? 1 : 'SIGTERM';
+const createKillMessage = prefix => new RegExp(
+    _.escapeRegExp(prefix) +
+    ' exited with code ' +
+    (isWindows ? 1 : '(SIGTERM|143)')
+);
 
 const run = args => {
     const child = spawn('node ./concurrently.js ' + args, {
@@ -222,7 +227,7 @@ describe('--kill-others', () => {
         child.log.pipe(buffer(child.close)).subscribe(lines => {
             expect(lines).toContainEqual(expect.stringContaining('[1] exit 0 exited with code 0'));
             expect(lines).toContainEqual(expect.stringContaining('Sending SIGTERM to other processes'));
-            expect(lines).toContainEqual(expect.stringContaining(`[0] sleep 10 exited with code ${killExitCode}`));
+            expect(lines).toContainEqual(expect.stringMatching(createKillMessage('[0] sleep 10')));
             done();
         }, done);
     });
@@ -232,7 +237,7 @@ describe('--kill-others', () => {
         child.log.pipe(buffer(child.close)).subscribe(lines => {
             expect(lines).toContainEqual(expect.stringContaining('[1] exit 0 exited with code 0'));
             expect(lines).toContainEqual(expect.stringContaining('Sending SIGTERM to other processes'));
-            expect(lines).toContainEqual(expect.stringContaining(`[0] sleep 10 exited with code ${killExitCode}`));
+            expect(lines).toContainEqual(expect.stringMatching(createKillMessage('[0] sleep 10')));
             done();
         }, done);
     });
@@ -242,7 +247,7 @@ describe('--kill-others', () => {
         child.log.pipe(buffer(child.close)).subscribe(lines => {
             expect(lines).toContainEqual(expect.stringContaining('[1] exit 1 exited with code 1'));
             expect(lines).toContainEqual(expect.stringContaining('Sending SIGTERM to other processes'));
-            expect(lines).toContainEqual(expect.stringContaining(`[0] sleep 10 exited with code ${killExitCode}`));
+            expect(lines).toContainEqual(expect.stringMatching(createKillMessage('[0] sleep 10')));
             done();
         }, done);
     });
@@ -263,7 +268,7 @@ describe('--kill-others-on-fail', () => {
         child.log.pipe(buffer(child.close)).subscribe(lines => {
             expect(lines).toContainEqual(expect.stringContaining('[1] exit 1 exited with code 1'));
             expect(lines).toContainEqual(expect.stringContaining('Sending SIGTERM to other processes'));
-            expect(lines).toContainEqual(expect.stringContaining(`[0] sleep 10 exited with code ${killExitCode}`));
+            expect(lines).toContainEqual(expect.stringMatching(createKillMessage('[0] sleep 10')));
             done();
         }, done);
     });
@@ -310,7 +315,7 @@ describe('--handle-input', () => {
         child.close.subscribe(exit => {
             expect(exit[0]).toBeGreaterThan(0);
             expect(lines).toContainEqual(expect.stringContaining('[1] stop'));
-            expect(lines).toContainEqual(expect.stringMatching(new RegExp(`\\[0\\] .*? ${killExitCode}`)));
+            expect(lines).toContainEqual(expect.stringMatching(createKillMessage('[0] node fixtures/read-echo.js')));
             done();
         }, done);
     });
@@ -328,7 +333,7 @@ describe('--handle-input', () => {
         child.close.subscribe(exit => {
             expect(exit[0]).toBeGreaterThan(0);
             expect(lines).toContainEqual(expect.stringContaining('[1] stop'));
-            expect(lines).toContainEqual(expect.stringMatching(new RegExp(`\\[0\\] .*? ${killExitCode}`)));
+            expect(lines).toContainEqual(expect.stringMatching(createKillMessage('[0] node fixtures/read-echo.js')));
             done();
         }, done);
     });

--- a/bin/concurrently.spec.js
+++ b/bin/concurrently.spec.js
@@ -6,8 +6,6 @@ const spawn = require('spawn-command');
 const isWindows = process.platform === 'win32';
 const killExitCode = isWindows ? 1 : 'SIGTERM';
 
-jest.setTimeout(3000);
-
 const run = args => {
     const child = spawn('node ./concurrently.js ' + args, {
         cwd: __dirname

--- a/bin/concurrently.spec.js
+++ b/bin/concurrently.spec.js
@@ -1,10 +1,12 @@
 const readline = require('readline');
 const Rx = require('rxjs');
-const { buffer, filter, map, takeUntil, tap } = require('rxjs/operators');
+const { buffer, map } = require('rxjs/operators');
 const spawn = require('spawn-command');
 
 const isWindows = process.platform === 'win32';
 const killExitCode = isWindows ? 1 : 'SIGTERM';
+
+jest.setTimeout(3000);
 
 const run = args => {
     const child = spawn('node ./concurrently.js ' + args, {
@@ -27,7 +29,12 @@ const run = args => {
         Rx.fromEvent(stderr, 'line')
     ).pipe(map(data => data.toString()));
 
-    return { close, log, stdin: child.stdin };
+    return {
+        close,
+        log,
+        stdin: child.stdin,
+        pid: child.pid
+    };
 };
 
 it('has help command', done => {
@@ -103,6 +110,17 @@ describe('exitting conditions', () => {
                 expect(exit[0]).toBeGreaterThan(0);
                 done();
             }, done);
+    });
+
+    it.skip('is of success when a SIGINT is sent', done => {
+        const child = run('"node fixtures/read-echo.js"');
+        child.close.subscribe(exit => {
+            // TODO This is null within Node, but should be 0 outside (eg from real terminal)
+            expect(exit[0]).toBe(0);
+            done();
+        }, done);
+
+        process.kill(child.pid, 'SIGINT');
     });
 
     it('is aliased to -s', done => {

--- a/src/flow-control/kill-on-signal.js
+++ b/src/flow-control/kill-on-signal.js
@@ -1,5 +1,6 @@
 const { map } = require('rxjs/operators');
 
+
 module.exports = class KillOnSignal {
     constructor({ process = global.process } = {}) {
         this.process = process;
@@ -18,9 +19,10 @@ module.exports = class KillOnSignal {
             const closeStream = command.close.pipe(map(value => {
                 return caughtSignal === 'SIGINT' ? 0 : value;
             }));
-
-            return Object.create(command, {
-                close: { get: () => closeStream }
+            return new Proxy(command, {
+                get(target, prop) {
+                    return prop === 'close' ? closeStream : target[prop];
+                }
             });
         });
     }

--- a/src/flow-control/kill-on-signal.js
+++ b/src/flow-control/kill-on-signal.js
@@ -1,15 +1,27 @@
+const { map } = require('rxjs/operators');
+
 module.exports = class KillOnSignal {
     constructor({ process = global.process } = {}) {
         this.process = process;
     }
 
     handle(commands) {
+        let caughtSignal;
         ['SIGINT', 'SIGTERM'].forEach(signal => {
             this.process.on(signal, () => {
+                caughtSignal = signal;
                 commands.forEach(command => command.kill(signal));
             });
         });
 
-        return commands;
+        return commands.map(command => {
+            const closeStream = command.close.pipe(map(value => {
+                return caughtSignal === 'SIGINT' ? 0 : value;
+            }));
+
+            return Object.create(command, {
+                close: { get: () => closeStream }
+            });
+        });
     }
 };

--- a/src/flow-control/kill-on-signal.spec.js
+++ b/src/flow-control/kill-on-signal.spec.js
@@ -13,8 +13,20 @@ beforeEach(() => {
     controller = new KillOnSignal({ process });
 });
 
-it('returns same commands', () => {
-    expect(controller.handle(commands)).toBe(commands);
+it('returns commands that map SIGINT to exit code 0', () => {
+    const newCommands = controller.handle(commands);
+    expect(newCommands).not.toBe(commands);
+    expect(newCommands).toHaveLength(commands.length);
+
+    const callback = jest.fn();
+    newCommands[0].close.subscribe(callback);
+    process.emit('SIGINT');
+
+    // A fake command's .kill() call won't trigger a close event automatically...
+    commands[0].close.next(1);
+
+    expect(callback).not.toHaveBeenCalledWith('SIGINT');
+    expect(callback).toHaveBeenCalledWith(0);
 });
 
 it('kills all commands on SIGINT', () => {

--- a/src/flow-control/restart-process.js
+++ b/src/flow-control/restart-process.js
@@ -41,8 +41,10 @@ module.exports = class RestartProcess {
                 return value === 0 || emission >= this.tries;
             }));
 
-            return Object.create(command, {
-                close: { get: () => closeStream }
+            return new Proxy(command, {
+                get(target, prop) {
+                    return prop === 'close' ? closeStream : target[prop];
+                }
             });
         });
     }

